pithos/pithos.py

```diff
@@ -877,11 +886,18 @@ def get_album_art(url, tmpdir, *extra):
                 return (None, None,) + extra
 
             file_url = None
+            song, index = extra
             if tmpdir:
                 try:
-                    with tempfile.NamedTemporaryFile(prefix='art-', suffix='.jpeg', dir=tmpdir.name, delete=False) as f:
-                        f.write(image)
-                        file_url = urllib.parse.urljoin('file://', urllib.parse.quote(f.name))
+                    # TODO move cache clear call somewhere more sane?
+                    self.clear_art_cache()
+                    encoded_filename = (song.artist+song.album).encode('utf-8').hex()[:222]+'.jpeg'
```

Comment:
I'm assuming the `:222` is to limit the length? Just use a hash like sha256 instead?